ratelimit.go

```diff
@@ -310,14 +310,14 @@ func (tb *Bucket) currentTick(now time.Time) int64 {
 // available in the bucket at the given time, which must
 // be in the future (positive) with respect to tb.latestTick.
 func (tb *Bucket) adjustavailableTokens(tick int64) {
+	tb.latestTick = tick
```

Comment:
Won't this cause the replenishment to fail?

`tb.availableTokens += (tick - tb.latestTick) * tb.quantum`
Will always increment by zero.